src/zernike_fet.cpp

```diff
@@ -295,6 +310,185 @@ void ZernikeFET::score_source(const BankedParticle& p) {
   }
 }
 
+double ZernikeFET::evaluate(const Position& r, const double& E) const {
+  StaticVector6 indices;
+  // get the energy-index, if energy-filter exists
+  if (energy_filter_) {
+    std::optional<std::size_t> E_indx = energy_filter_->get_index(E);
+    if (E_indx.has_value() == false) {
+      // Not inside any energy bin. return 0.;
+      return 0.;
+    }
+    indices.push_back(E_indx.value());
+  }
+  // create a temperaroy tracker for getting a position index
+  Direction dir;
+  Tracker trkr(r, dir);
+  // get the positional indexes from cylinder_filter
+  StaticVector3 cylinder_index = cylinder_filter_->get_indices(trkr);
```

Comment:
Maybe for cylinder filters and cartesian filters, we can create a special method that can return the indices with just the position ? That would allow us to avoid the creation of this temporary Tracker, which should greatly reduce the computational costs. Because creating a tracker in this case is a lot of computational work and it is not purely necessary for these two types of position filters. We couldn't use this on a general tally, as those could have any type of position filter, but here we know what we are getting. This could also be done for the LegendreFET.